packages/react-data-grid/src/RowGroup.js

```diff
@@ -71,35 +66,37 @@ RowGroup.propTypes = {
   renderer: PropTypes.func
 };
 
-const  DefaultRowGroupRenderer = (props) => {
-  let treeDepth = props.treeDepth || 0;
-  let marginLeft = treeDepth * 20;
-
-  let style = {
-    height: '50px',
-    border: '1px solid #dddddd',
-    paddingTop: '15px',
-    paddingLeft: '5px'
-  };
-
-  let onKeyDown = (e) => {
-    if (e.key === 'ArrowLeft') {
-      props.onRowExpandToggle(false);
-    }
-    if (e.key === 'ArrowRight') {
-      props.onRowExpandToggle(true);
-    }
-    if (e.key === 'Enter') {
-      props.onRowExpandToggle(!props.isExpanded);
-    }
-  };
-  return (
-    <div style={style} onKeyDown={onKeyDown} tabIndex={0}>
-      <span className="row-expand-icon" style={{float: 'left', marginLeft: marginLeft, cursor: 'pointer'}} onClick={props.onRowExpandClick} >{props.isExpanded ? String.fromCharCode('9660') : String.fromCharCode('9658')}</span>
-      <strong>{props.columnGroupName}: {props.name}</strong>
-    </div>
-  );
-};
+class DefaultRowGroupRenderer extends React.PureComponent {
+  render() {
+    let treeDepth = this.props.treeDepth || 0;
+    let marginLeft = treeDepth * 20;
+
+    let style = {
+      height: '50px',
+      border: '1px solid #dddddd',
+      paddingTop: '15px',
+      paddingLeft: '5px'
+    };
+
+    let onKeyDown = (e) => {
```

Comment:
I'm guessing this really doesn't matter because we are only rendering a <div> but if I'm wrong it may be worth moving the handler into a method so it is the same function instance each time we render? Similarly, the style object is already constant so we could define it as such.

Also, no action here right now, but we probably only ever use treeDepth 0 and 1 to my knowledge. As such I'm guessing it should always be 0 here (which would make this a completely dumb component)? Probably not worth making a change about, but worth thinking about since similar specifics of our use case may have more impact elsewhere.

Reply:
@mkohlmyr I would summarise your comment as "there is plenty more stuff to delete and/or improve", which I entirely agree with :)

Since the specific things you mention are existing issues I'm not going to do anything about them now, but I will definitely be coming back to remove more stuff in the future.